Channel/DoctrineChannel.php

```diff
@@ -43,20 +46,38 @@ public function supports($recipient)
      */
     public function configure(OptionsResolver $resolver)
     {
+        $resolver
+            ->setDefined(['attachments_path'])
+        ;
     }
 
     /**
      * @inheritdoc
      */
     public function handle(Delivery $delivery)
     {
+        $options = $delivery->getOptions();
+
         $notification = new Notification(
             $delivery->getSubject(),
             $delivery->getBody(),
             $delivery->getRecipient()
         );
 
+        foreach ($delivery->getAttachments() as $attachment) {
+            /** @var $attachment File */
```

Comment:
If the type in your phpdoc is correct (@param File [] $attachments), this line should not exists